### PR TITLE
UI: Use FloatingViewportContent in widgets

### PR DESCRIPTION
**Pull request by NancyMcCallB**
Make small fixes to FloatingViewportContent so that it properly updates active content when the it's hosted by a widget. Also, add ViewportWidget to appui-test-providers to have a test/sample for hosting the FloatingViewportContent in a widget.

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

**Review by raplemie — APPROVED** [review]
Realized late that I made my comments on test providers, so they dont have much of an impact. Only thing I'd like back is the standalone (chome) launch.

File: .vscode/launch.json [hunk 1]
```diff
@@ -1479,17 +1522,28 @@
...
+    {
+      "name": "appui-test-app/standalone (electron)",
       "presentation": {
         "group": "1_TestApps",
         "order": 1
       },
       "configurations": [
-        "[BACKEND] appui-test-app/standalone (chrome)",
-        "[FRONTEND] appui-test-app/standalone (chrome)"
+        "[BACKEND] appui-test-app/standalone (electron)",
```
[the remark this answers is not in the document]
Comment (NancyMcCallB) [reply .vscode/launch.json:1586]: Restored!

File: test-apps/appui-test-app/appui-test-providers/src/ui/widgets/ViewportWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    setTimeout(() => widgetDef?.setWidgetState(WidgetState.Floating));
+  }, [widgetDef]);
+
+  React.useEffect(() => {
+    async function setupView() {
+      if (undefined === viewState && activeIModelConnection) {
+        const defaultViewId = await activeIModelConnection?.views?.queryDefaultViewId();
+        if (defaultViewId && Id64.isValidId64(defaultViewId)) {
+          const newViewState = await activeIModelConnection?.views.load(defaultViewId);
+          // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (NancyMcCallB) [reply test-apps/appui-test-app/appui-test-providers/src/ui/widgets/ViewportWidget.tsx:31]: Yes, it complains if viewState is not a dependency.

File: test-apps/appui-test-app/appui-test-providers/src/ui/components/ViewDefinitionSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  React.useEffect(() => {
+    setAvailableViewDefinitions([]);
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    getViewDefinitions(props.imodel).then(setAvailableViewDefinitions);
+  }, [props.imodel]);
+  const onViewDefinitionSelected = props.onViewDefinitionSelected;
+  const memoizedOnViewDefinitionSelected = React.useCallback((selectedId: string) => {
+    if (onViewDefinitionSelected)
+      onViewDefinitionSelected(selectedId);
+  }, [onViewDefinitionSelected]);
```
[the remark this answers is not in the document]
Comment (NancyMcCallB) [reply test-apps/appui-test-app/appui-test-providers/src/ui/components/ViewDefinitionSelector.tsx]: Changed it to remove React.useCallback and use the prop directly.

File: test-apps/appui-test-app/appui-test-providers/src/ui/components/ViewDefinitionSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const selectOptions = React.useMemo<SelectOption<string>[]>(() => {
+    return (availableViewDefinitions ?? []).map((definition) => (
+      { value: definition.id, label: definition.label }
+    ));
+  }, [availableViewDefinitions]);
```
[the remark this answers is not in the document]
Comment (NancyMcCallB) [reply test-apps/appui-test-app/appui-test-providers/src/ui/components/ViewDefinitionSelector.tsx]: Reworked this code to store the SelectOptions in state and set them directly from the useEffect hook.

File: .vscode/launch.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (raplemie) [reply .vscode/launch.json:1586]: Thanks!

**Pull request merged**